packages/apputils-extension/src/statusbarplugin.ts

```diff
@@ -188,25 +195,66 @@ export const runningSessionsStatus: JupyterFrontEndPlugin<void> = {
     if (settingRegistry) {
       let disposable: IDisposable;
       const onSettingsUpdated = (
-        settings: ISettingRegistry.ISettings
+        kernelSettings?: ISettingRegistry.ISettings,
+        terminalsSettings?: ISettingRegistry.ISettings
       ): void => {
-        const showStatusBarItem = settings.get('showStatusBarItem')
-          .composite as boolean;
+        const showTerminalsMap = {
+          'if-any': undefined,
+          never: false,
+          always: true
+        };
+        const showKernels =
+          (kernelSettings?.get('showStatusBarItem').composite as
+            | boolean
+            | undefined) ?? true;
+        const showTerminals =
+          showTerminalsMap[
+            (terminalsSettings?.get('showStatusBarItem').composite as
+              | 'if-any'
+              | 'never'
+              | 'always'
+              | undefined) ?? 'if-any'
+          ];
 
         disposable?.dispose();
-        if (showStatusBarItem) {
-          disposable = registerItem();
+        if (showKernels || showTerminals) {
```

Comment:
```suggestion
        if (showKernels || showTerminals !== false) {
```

Reply:
As a reminder, for a future PR with the token approach, we could define the `showTerminals` as `boolean | 'if-any'` to clarify the state, instead of using `underfined` as a third boolean state.